packages/forms-web-app/src/routes/file-based-router/comment-planning-appeal/your-name/controller.js

```diff
@@ -0,0 +1,77 @@
+/** @type {import('express').RequestHandler} */
+const yourNameGet = (req, res) => {
+	if (!req.session.interestedParty?.appealNumber) {
+		return res.redirect(`enter-appeal-reference`);
+	}
+
+	const interestedParty = req.session.interestedParty || {};
+
+	res.render(`comment-planning-appeal/your-name/index`, { interestedParty });
+};
+
+/** @type {import('express').RequestHandler} */
+const yourNamePost = async (req, res) => {
+	const { 'first-name': firstName, 'last-name': lastName } = req.body;
+
+	req.session.interestedParty.firstName = firstName;
+	req.session.interestedParty.lastName = lastName;
+
+	let errors = {};
+	let errorSummary = [];
+
+	if (!firstName) {
+		errors['first-name'] = {
+			msg: 'Enter your first name',
```

Comment:
might be nice to wrap all this stuff into a piece of middleware like the email validator. Could be a generic string validator that takes fieldName and inputLength as options?

Reply:
Good point, will make it nicer - will have a refactor